### PR TITLE
[AArch64] Allow SVE code generation for fixed-width vectors

### PR DESCRIPTION
**Pull request by igogo-x86**
This patch implicitly decreases the cost of masked load and stores with fixed width vectors when their width is equal to 128 bytes. Also, it allows the generation of SVE code with masks that mimic Neon.

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/sve-fixed-length-masked-128bit-loads.ll [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+; CHECK-NEXT:    cmlt v0.4s, v0.4s, #0
+; CHECK-NEXT:    cmpne p0.s, p0/z, z0.s, #0
+; CHECK-NEXT:    ld1w { z0.s }, p0/z, [x0]
+; CHECK-NEXT:    // kill: def $q0 killed $q0 killed $z0
+; CHECK-NEXT:    ret
+  %load = call <4 x float> @llvm.masked.load.v4f32(ptr %src, i32 8, <4 x i1> %mask, <4 x float> zeroinitializer)
+  ret <4 x float> %load
+}
+
+define <2 x double> @masked_load_v2f64(ptr %src, <2 x i1> %mask) {
```
Comment (david-arm) [inline llvm/test/CodeGen/AArch64/sve-fixed-length-masked-128bit-loads.ll:55]: Is it worth having at least one test where the passthru is not zero? i.e.

define <4 x float> @masked_load_v4f32(ptr %src, <4 x i1> %mask, <4 x float> %passthru) {
  %load = call <4 x float> @llvm.masked.load.v4f32(ptr %src, i32 8, <4 x i1> %mask, <4 x float> %passthru)
   ret <4 x float> %load
}

Comment (igogo-x86) [reply llvm/test/CodeGen/AArch64/sve-fixed-length-masked-128bit-loads.ll:55]: Added `masked_load_passthru_v2f64`

**Review by david-arm — APPROVED** [review]
LGTM! Thanks @igogo-x86. :)

**Comment by nikic** [comment]
Pushed https://github.com/llvm/llvm-project/commit/e3adc6a1f7ddba9475940205eb55f2258c71aa84 as a speculative build fix.

**Pull request merged**